tests/test_jsonyamlxml_writer.py

```diff
@@ -1,36 +1,74 @@
-    os.remove(temporary_file_path)
+    file_with_ending = glob.glob(temporary_file_path + "*")
+    for file in file_with_ending:
+        os.remove(file)
 
 
-def test_external_package_references(temporary_file_path) -> None:
+@pytest.mark.parametrize("out_format", ['yaml', 'xml', 'json'])
+def test_external_package_references(temporary_file_path: str, out_format: str) -> None:
     document: Document = minimal_document()
     package: Package = document.packages[0]
     first_ref = ExternalPackageRef(category="PACKAGE-MANAGER")
     second_ref = ExternalPackageRef(category="SECURITY")
     package.add_pkg_ext_refs(first_ref)
     package.add_pkg_ext_refs(second_ref)
 
-    with open(temporary_file_path, "w") as out:
-        json.write_document(document, out, validate=False)
+    file_path_with_ending = temporary_file_path + "." + out_format
+    write_anything.write_file(document, file_path_with_ending, validate=False)
 
-    parsed_document = parse_file(temporary_file_path)[0]
+    parsed_document = parse_file(file_path_with_ending)[0]
 
     written_package: Package = parsed_document.packages[0]
     assert len(written_package.pkg_ext_refs) is 2
     written_reference_categories = list(map(lambda x: x.category, written_package.pkg_ext_refs))
     assert first_ref.category in written_reference_categories
     assert second_ref.category in written_reference_categories
+
+
+@pytest.mark.parametrize("out_format", ['yaml', 'xml', 'json'])
+def test_primary_package_purpose(temporary_file_path: str, out_format: str):
+    document: Document = minimal_document()
+    package: Package = document.packages[0]
+    package.primary_package_purpose = PackagePurpose.FILE
+
+    file_path_with_ending = temporary_file_path + "." + out_format
+    write_anything.write_file(document, file_path_with_ending, validate=False)
+
+    parsed_document: Document = parse_file(file_path_with_ending)[0]
+    written_package: Package = parsed_document.packages[0]
+
+    assert written_package.primary_package_purpose == PackagePurpose.FILE
+
+@pytest.mark.parametrize("out_format", ['yaml', 'xml', 'json'])
+def test_release_built_valid_until_date(temporary_file_path: str, out_format: str):
+    document: Document = minimal_document()
+    package: Package = document.packages[0]
+    package.release_date = datetime(2021, 1, 1, 12, 0, 0)
+    package.built_date = datetime(2021, 1, 1, 12, 0, 0)
+    package.valid_until_date = datetime(2022, 1, 1, 12, 0, 0)
+
+    file_path_with_ending = temporary_file_path + "." + out_format
+    write_anything.write_file(document, file_path_with_ending, validate=False)
+
+    parsed_document: Document = parse_file(file_path_with_ending)[0]
+    written_package: Package = parsed_document.packages[0]
+
+    assert written_package.release_date == datetime(2021, 1, 1, 12, 0, 0)
+    assert written_package.built_date == datetime(2021, 1, 1, 12, 0, 0)
+    assert written_package.valid_until_date == datetime(2022, 1, 1, 12, 0, 0)
```

Comment:
Better to compare against the values `package.release_date` etc

Reply:
fixed